integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingSenderTestCase.java

```diff
@@ -0,0 +1,246 @@
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("400"), "Expected 400, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 400 Bad Request"),
+                "Required payload not found");
+    }
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking unauthorized http status endpoint with blocking call", enabled = true)
+    public void testUnauthorizedHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("UnauthorizedEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("401"), "Expected 401, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 401 Unauthorized"),
+                "Required payload not found");
+    }
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking forbidden http status endpoint with blocking call", enabled = true)
+    public void testForbiddenHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("ForbiddenEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("403"), "Expected 403, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 403 Forbidden"),
+                "Required payload not found");
+    }
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking not found http status endpoint with blocking call", enabled = true)
+    public void testNotFoundHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("NotFoundEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("404"), "Expected 404, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 404 Not Found"),
+                "Required payload not found");
```

Comment:
Most of the code lines in the test methods are equal. Can we extract a method and reuse it in every place?

Reply:
Done in commit [f23167f](https://github.com/wso2/product-ei/pull/3468/commits/f23167fede3bebc4cb3dba75083e82c95eab860c)